### PR TITLE
Ozone Bid Adapter: added or updated support for multiple modules

### PR DESCRIPTION
**Pull request by afsheenb**
- [X] Feature

## Description of change

*  full adapter aliasing support
*  updated support on Parrable and pubCommonId module
*  added support for Lotame, Neustar and SharedId user modules

Contact:
- engineering@ozoneproject.com

### CODE REVIEW

**Comment by afsheenb** [comment]
> This pull request **introduces 1 alert** when merging [dcccf18](https://github.com/prebid/Prebid.js/commit/dcccf18bd00d0287bffcbd08f0e54e1ee641872b) into [26a268a](https://github.com/prebid/Prebid.js/commit/26a268a0899634998f7ccc05243238c55c389d84) - [view on LGTM.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-5ddc9c84eeb2060eda481bf71f4b53df6c1f0498)
> 
> **new alerts:**
> 
>     * 1 for Useless assignment to local variable

Pushed a fix for this just now.

**Review by Fawke — COMMENTED** [review]
Hi @afsheenb,

Can you please add some details on the modules that your updating in the PR. That'll help me concentrate on those changes while doing the review since I see quite a lot of changes in the JS file.

**Comment by AskRupert-DM** [comment]
> Hi @afsheenb,
> 
> Can you please add some details on the modules that your updating in the PR. That'll help me concentrate on those changes while doing the review since I see quite a lot of changes in the JS file.

Hey @Fawke what details do you require - the modules we've added support for are Lotame, Neustar and SharedId user modules. We also cleaned up the code that is used for pubCommonId and ParrableId.

The reason for the file looking quite different is because of the alias support we've built into the adapter - things like logs have been updated to reference the alias name thats in use instead of 'ozone'.

I have a testpage here if you are interested in seeing it working - https://www.ardm.io/ozone/2.5.0/final-qa/ozone-test.html

**Comment by Fawke** [comment]
@afsheenb,

Thanks for the clarification, I think this change looks good. Can you also create a docs PR to add these new userId modules to your list of supported modules?

Here's an excerpt for PR guidelines:

```
Below are some examples of bidder specific updates that should require docs update (in their dev-docs/bidders/BIDDER.md file):
If they support the GDPR consentManagement module and TCF1, add gdpr_supported: true
If they support the GDPR consentManagement module and TCF2, add tcf2_supported: true
If they support the US Privacy consentManagementUsp module, add usp_supported: true
If they support one or more userId modules, add userId: (list of supported vendors)
If they support video and/or native mediaTypes add media_types: video, native. Note that display is added by default. If you don't support display, add "no-display" as the first entry, e.g. media_types: no-display, native
If they support COPPA, add coppa_supported: true
If they support SChain, add schain_supported: true
If their bidder doesn't work well with safeframed creatives, add safeframes_ok: false. This will alert publishers to not use safeframed creatives when creating the ad server entries for their bidder.
If they're setting a deal ID in some scenarios, add bidder_supports_deals: true
If they have an IAB Global Vendor List ID, add gvl_id: ID. There's no default.
```

Maybe you wanna add `gvl_id` as well to the docs, since you added one in this PR. Updating the docs helps some publishers to quickly see the features of your adapter, that's why it's encouraged to it updated. Thanks.

**Comment by afsheenb** [comment]
Thanks @Fawke ! An update to dev-docs/bidders/ozone.md has been submitted in this PR: https://github.com/prebid/prebid.github.io/pull/2716 .

**Review by Fawke — APPROVED** [review]

File: modules/ozoneBidAdapter.js [hunk 1]
```diff
@@ -535,26 +601,31 @@ export const spec = {
...
         }
       }
       var lipbid = utils.deepAccess(bidRequest.userId, 'lipb.lipbid');
       if (lipbid) {
         ret['lipb'] = {'lipbid': lipbid};
       }
-      var id5id = utils.deepAccess(bidRequest.userId, 'id5id.uid');
-      if (id5id) {
-        ret['id5id'] = id5id;
-      }
```
Comment (smenzer) [inline modules/ozoneBidAdapter.js:557]: this should not have been removed; or the code above needs to handle the fact that the id5id is an object not a string

File: test/spec/modules/ozoneBidAdapter_spec.js [hunk 2]
```diff
@@ -63,10 +67,83 @@ var validBidRequestsWithUserIdData = [
...
+    params: { publisherId: '9876abcd12-3', customData: [{'settings': {}, 'targeting': {'gender': 'bart', 'age': 'low'}}], placementId: '1310000099', siteId: '1234567890', id: 'fea37168-78f1-4a23-a40e-88437a99377e', auctionId: '27dcb421-95c6-4024-a624-3c03816c5f99', imp: [ { id: '2899ec066a91ff8', tagid: 'undefined', secure: 1, banner: { format: [{ w: 300, h: 250 }, { w: 300, h: 600 }], h: 250, topframe: 1, w: 300 } } ] },
     sizes: [[300, 250], [300, 600]],
     transactionId: '2e63c0ed-b10c-4008-aed5-84582cecfe87',
-    userId: {'pubcid': '12345678', 'id5id': { 'uid': 'ID5-someId' }, 'criteortus': {'ozone': {'userid': 'critId123'}}, 'idl_env': 'liverampId', 'lipb': {'lipbid': 'lipbidId123'}, 'parrableId': {eid: 'parrableid123'}}
+    userId: {
+      'pubcid': '12345678',
+      'tdid': '1111tdid',
+      'id5id': 'ID5-someId',
```
Comment (smenzer) [inline test/spec/modules/ozoneBidAdapter_spec.js:76]: this is not what the `id5id` object looks like, this line should be:
```
'id5id': { 'uid': 'ID5-someId' }
```

File: test/spec/modules/ozoneBidAdapter_spec.js [hunk 3]
```diff
@@ -272,32 +327,10 @@ var validBidderRequest1OutstreamVideo2020 = {
...
-                  }
-                ]
-              }
-            }
-          }
+          ]
         },
         'userId': {
-          'id5id': { uid: 'ID5-ZHMOpSv9CkZNiNd1oR4zc62AzCgSS73fPjmQ6Od7OA' },
+          'id5id': 'ID5-ZHMOpSv9CkZNiNd1oR4zc62AzCgSS73fPjmQ6Od7OA',
```
Comment (smenzer) [inline test/spec/modules/ozoneBidAdapter_spec.js:333]: this line should not have been changed

File: test/spec/modules/ozoneBidAdapter_spec.js [hunk 4]
```diff
@@ -2118,13 +2016,14 @@ describe('ozone Adapter', function () {
       bidRequests[0]['userId'] = {
         'criteortus': '1111',
         'digitrustid': {data: {id: 'DTID', keyv: 4, privacy: {optout: false}, producer: 'ABC', version: 2}},
-        'id5id': {'uid': '2222'},
+        'id5id': '2222',
```
Comment (smenzer) [inline test/spec/modules/ozoneBidAdapter_spec.js:2019]: this line should not have been changed

File: test/spec/modules/ozoneBidAdapter_spec.js [hunk 5]
```diff
@@ -2138,62 +2037,121 @@ describe('ozone Adapter', function () {
...
```
Comment (smenzer) [inline test/spec/modules/ozoneBidAdapter_spec.js:2040]: this line should not have been changed

File: test/spec/modules/ozoneBidAdapter_spec.js [hunk 6]
```diff
@@ -2138,62 +2037,121 @@ describe('ozone Adapter', function () {
...
       delete validBidRequests[0].userId; // tidy up now, else it will screw with other tests
     });
 
-    it('should add a user.ext.eids object to contain user ID data in the new location (Nov 2019)', function() {
+    it('should add a user.ext.eids object to contain user ID data in the new location (Nov 2019) Updated Aug 2020', function() {
       const request = spec.buildRequests(validBidRequestsWithUserIdData, validBidderRequest.bidderRequest);
+      /*
+      'pubcid': '12345678',
+      'tdid': '1111tdid',
+      'id5id': 'ID5-someId',
```
Comment (smenzer) [inline test/spec/modules/ozoneBidAdapter_spec.js:2059]: this is not what the `id5id` object looks like, this line should be:
```
'id5id': { 'uid': 'ID5-someId' }
```

**Review by smenzer — COMMENTED** [review]
i know this has already been merged, but it should not have been, given that the ID5 ID is not read correctly in the bid adapter and the tests are using an incorrect value to test against (which is why they passed).

@afsheenb can you please create a new PR addressing the points I commented on?

**Comment by AskRupert-DM** [comment]
Hey @smenzer - I'll get this updated in the next release - the support of ID5s module was done for future-proofing purposes as opposed to it being used by publishers on our end actively - either way we will ensure its updated accordingly in our next release.

**Comment by smenzer** [comment]
Hi @AskRupert-DM -- thank you!

**Comment by patmmccann** [comment]
@AskRupert-DM please note LGTM error introduced on this PR

https://github.com/prebid/Prebid.js/pull/6324#issuecomment-781330952

**Pull request merged**